### ISSUE TITLE
Connection fails without version parameter specified.

### ISSUE THREAD

**Issue by porco-rosso-j** [MetaMask/template-snap-monorepo#168]
I suggest to pass version to `useRequestSnap` like below in [index.tsx](https://github.com/MetaMask/template-snap-monorepo/blob/main/packages/site/src/pages/index.tsx) as it would fail to connect otherwise. 

```
  const requestSnap = useRequestSnap(defaultSnapOrigin, '0.1.0');
```

**Comment by porco-rosso-j** [comment]
Plus, calling the `hello` method doesn't work, throwing the following RPC error.  I encountered this with a snap where I upgraded dependencies to the latest, so probably due to new updates. Would be nice if anybody knows walkaround. 

```
 Expected the value to satisfy a union of `record | array`, but received: undefined
```

![Screenshot 2024-03-11 at 1 57 40](https://github.com/MetaMask/template-snap-monorepo/assets/88586592/7154502c-cdf8-4ea1-b168-0ed26e08057d)
